### ISSUE TITLE
Elementor Lightbox navigation (prev/next) in native WordPress Gutenberg gallery block

### ISSUE THREAD

**Issue by euene** [elementor/elementor#11942]
When the Elementor Image Lightbox option is enabled and a gallery is inserted using the native WordPress Gutenberg gallery block (with the setting "Link to: Media FIle") - the full-size images open in Elementor's lightbox after clicking the thumbnail, which is great.

![lightbox](https://user-images.githubusercontent.com/6027160/88318644-b310e600-cd23-11ea-944e-57a3080038aa.png)

However, this still doesn't function like a proper "gallery" as there is no navigation/previous/next arrows to go between images. It would be great if Elementor can detect the native Gutenberg gallery and add arrows navigation like its own Elementor galleries.

The Gutenberg gallery block produces a markup similar to this for galleries, so perhaps the Elementor lightbox can look for it and construct the gallery list from it.

```
<ul class="blocks-gallery-grid">
    <li class="blocks-gallery-item">
        <figure>
            <a href="full-1.jpg">
                <img src="thumbnail-1.jpg" />
            </a>
        </figure>
    </li>
    <li class="blocks-gallery-item">
        <figure>
            <a href="full-2.jpg">
                <img src="thumbnail-2.jpg" />
            </a>
        </figure>
    </li>
</ul>
```

- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**Comment by vovkasolovev** [comment]
Please make navigation between images in the Gutenberg gallery if the gallery is shown via Elementor Lightbox.

**Comment by dasigna** [comment]
CRAZY. another one of these issues thats as well as annoying as it is  l o o o o n g  overdue ...

any chance to get this fixed please?

**Comment by magneting** [comment]
Same problem here :(

**Comment by dgdb1** [comment]
I have the same problem.

**Comment by ruefe** [comment]
Yeah same problem here… been looking for this already a year ago and hoped that there would be a solution now…

**Comment by euene** [comment]
As a workaround before Elementor fixes this in core, you can add the following script at the end of `<body>` which will add the necessary data attribute to enable the gallery navigation.

```html
<script>
jQuery(document).ready(function($) {
    $('figure.wp-block-gallery').each(function(index) {
        $(this).find('a').attr('data-elementor-lightbox-slideshow', 'elementor-wp-block-gallery-' + index);
    });
});
</script>
```

**Comment by ruefe** [comment]
Wow this works perfectly! Thank you so much!

**Comment by Florian4573** [comment]
Hi iam not a Pro like you so where should i add this script - in which file? 

Thanks for helping!

Best regards
Florian

**Comment by ruefe** [comment]
Easiest way is to add the above mentioned lines of code via a HTML-Widget in Elementor. I added it in a Single-Page Template, but i think you could also add it in the Footer or Header Template, so that it will be active on every Content-page.

**Comment by euene** [comment]
You shouldn't add the script via the Elementor template, instead you should do it via the Custom Code feature of Elementor Pro (or any other plug-in which allows you to add scripts at the end of the body if not using Elementor Pro and only using Elementor Free).

https://elementor.com/help/custom-code-pro/

You should select body end as the location and make sure that the load jQuery checkbox is selected.

**Comment by Florian4573** [comment]
Many thanks it works perfectly!

**Comment by CarlosSouza** [comment]
@bratvanov Thank you for your solution!

It almost worked perfectly, the only issue is that sometimes galleries have caption which may or not include links, so we need to be more specific and target image links only:

`$('figure.wp-block-gallery').each(function(index) {
    $(this).find('.wp-block-image a').attr('data-elementor-lightbox-slideshow', 'elementor-wp-block-gallery-' + index);
});`

**Comment by jadamec** [comment]
Both of the "solutions" don't work for me. Good old WP Bakery...

**Comment by luccashoff** [comment]
Same for me. Paste both of these codes in the Custom Code and it didn't work for me.
Navigation tools keep appearing in the bottom of the page and it doesn't work
